docker/wsgi.py

```diff
@@ -13,15 +13,26 @@
 import json
 import os
 from mms.arg_parser import ArgParser
-
+import fasteners
 mms_arg_header = 'MMS Argument'
 args = []
 found_mms_args = 0
 
 mms_config_path = os.environ['MXNET_MODEL_SERVER_CONFIG']
 is_gpu_image = os.environ['MXNET_MODEL_SERVER_GPU_IMAGE']
-
-
+num_gpu = 1
+LOCK_FILE = '/tmp/tmp_lock_file'
+@fasteners.interprocess_locked(LOCK_FILE)
+def get_gpu_ids(args, n_gpu):
+    args.append('--gpu')
+    gpu_id=0
+    n_gpu =4
+    with open('/mxnet_model_server/.gpu_id', 'r') as file:
+        gpu_id = file.read()
+        args.append(str(int(gpu_id )% n_gpu))
+    with open('/mxnet_model_server/.gpu_id', 'w') as file:
+        file.write(str(int(gpu_id) + 1))
```

Comment:
Why open the same file twice? Why not open it with "r+"

Reply:
r+ appends the file and does not overwrite